op-node/rollup/sequencing/origin_selector_test.go

```diff
@@ -189,6 +196,28 @@ func TestOriginSelectorAdvances(t *testing.T) {
 	next, err = s.FindL1Origin(ctx, l2Head)
 	require.Nil(t, err)
 	require.Equal(t, c, next)
+
+	// Now force the retrieval of the next L1 origin
+	l1.ExpectL1BlockRefByHash(c.Hash, c, nil)
+	// It may not be ready yet. Expect an error, even if we could stick to c.
+	l1.ExpectL1BlockRefByNumber(d.Number, eth.BlockRef{}, ethereum.NotFound)
+
+	// at origin d, and can stick to c (thanks to timestamp), if it wasn't for recover mode.
+	s.recoverMode = true
+	l2Head = eth.L2BlockRef{
+		L1Origin: c.ID(),
+		Time:     d.Time + 4,
+	}
+	_, err = s.FindL1Origin(ctx, l2Head)
+	require.ErrorIs(t, err, derive.ErrTemporary)
+	require.ErrorIs(t, err, ethereum.NotFound)
+
+	// Now actually get to L1 origin d.
+	l1.ExpectL1BlockRefByHash(c.Hash, c, nil)
+	l1.ExpectL1BlockRefByNumber(d.Number, d, nil)
+	next, err = s.FindL1Origin(ctx, l2Head)
+	require.Nil(t, err)
```

Comment:
```suggestion
	require.NoError(t, err)
```
I know, we do a lot of `Nil` assertions on errors, but `NoError` is better at handling some edge cases, and prints a better message if the assertion fails.